documentation/source/requirements.txt

```diff
@@ -4,3 +4,4 @@ sphinx_rtd_theme
 sphinx-copybutton
 furo==2023.09.10
 sphinx
+sphinx_argparse
```

Comment:
In this week's SCT meeting, feedback was raised to see if we can get `sphinx_argparse` to create linkable anchors for each of the arguments in a script's argparse help.

Reply:
I've tried, but I was unable to make this work. :(

- `sphinx-argparse` uses `docutils.nodes` under the hood, and creates a tree structure where each element of the documentation has a specific node type. 
- The anchor links for argument section titles (`MANDATORY ARGUMENTS`, etc.) are generated via the `nodes.title` type.
  - Title nodes get converted into this `headerlink` HTML: 
     ```
     <h2>MANDATORY ARGUMENTS<a class="headerlink" href="[#mandatory-arguments](view-source:http://localhost:63342/spinalcordtoolbox/documentation/build/html/user_section/command-line/sct_deepseg.html#mandatory-arguments)" title="Link to this heading">¶</a></h2>
     ```
  - This transformation from "docutils node" to "formatted html" is _not_ handled by the extension itself. Instead, it's handled by Sphinx in the [`add_permalink_ref`](https://github.com/sphinx-doc/sphinx/blob/914f3f4e541309378d506662d2dd4b392f3691f2/sphinx/writers/html5.py#L413-L418) method, which gets called by the `depart_title` method for titles.
- By comparison, each individual argparse argument is given the `nodes.option_list_item` type instead (which makes sense!). 
  - Option list nodes get converted into this html:
     ```
     <dl class="option-list">
     <dt><kbd>-i</kbd></dt>
     <dd><p>Image to segment. Can be multiple images (separated with space).</p>
     </dd> 
     [...]
     </dl>
     ```
  - The `<kbd>` tag corresponds to the [`depart_literal`](https://github.com/sphinx-doc/sphinx/blob/914f3f4e541309378d506662d2dd4b392f3691f2/sphinx/writers/html5.py#L619-L621) method, which has no such options for headerlinks.

So, unless we somehow modify the `sphinx_argparse` extension to _not_ generate option lists, or somehow modify `sphinx` to change how it handles option lists, I think we're out of luck here.